web-client/dist/nodejs/worker.mjs

```diff
@@ -9,6 +9,24 @@ import wasm from './worker-wasm/index.js';
 
 // Provide a global WebSocket implementation, which is expected by the WASM code built for browsers.
 global.WebSocket = websocket.w3cwebsocket;
+// Workaround for libp2p's websocket-websys transport's worker detection,
+// which checks for web workers' `WorkerGlobalScope` interface.
```

Comment:
Per our discussion it seems like this is more a workaround to `nodejs` not supporting the `web-api` rather than a workaround to libp2p's websocket-websys transport's worker detection.